### ISSUE TITLE
twrp fails to decrypt on Moto G4 Plus (athene)

### ISSUE THREAD

**Issue by theScrabi** [TeamWin/Team-Win-Recovery-Project#1312]
- [x] I am running an official build of TWRP, downloaded from https://twrp.me/Devices/
- [x] I am running the latest version of TWRP
- [x] I have read the FAQ (https://twrp.me/FAQ/)
- [x] I have searched for my issue and it does not already exist

I can not decrypt /data on my athene phone. Now it seems, that this issue got fixed with twrp 3.2.3-1 however this is not yet available for the athene version of the G4 plus. The latest version here is twrp-3.2.1-0. So id like to ask to compile and upload 3.2.3-1 for athene as well :)

**Comment by pehlm** [comment]
Hope the same! But the Athene TWRP project don't seems to be maintained anymore: https://jenkins.twrp.me/job/athene-prod/.

**Comment by theScrabi** [comment]
What can we do to push it anyway?

**Comment by pehlm** [comment]
Maybe they stopped because of the 32 bit vs 64 bit debate for Athene, don't know. A 64 bit 3.2.2 build of TWRP is here though: https://forum.xda-developers.com/moto-g4-plus/development/recovery-twrp-3-2-1-unofficial-touch-t3722206, but reading in the comments it has problems for 32 bit. According to the developer his recovery build should work for that also, but I don't know, I haven't tried it, I use the 32 bit OmniROM for Athene, though, so I don't want to try.

But this project haven't yet 3.2.3 neither so....

**Comment by hansenhans** [comment]
Same problem on Wileyfox Swift (crackling). Decryption pattern works when booting LineageOS 15.1, but when I boot into TWRP, it doesn't accept the pattern. Everything worked before with LO 14.1 and TWRP 3.1.x (?)

**Comment by CaptainThrowback** [comment]
> Same problem on Wileyfox Swift (crackling). Decryption pattern works when booting LineageOS, but when I boot into TWRP, it doesn't accept the pattern.

Is your pattern the standard 3x3, or is it one of the custom Lineage patterns like 4x4 or 5x5, etc? I noticed a similar issue with larger-than-standard patterns and TWRP decrypt not working.

**Comment by hansenhans** [comment]
> > Same problem on Wileyfox Swift (crackling). Decryption pattern works when booting LineageOS, but when I boot into TWRP, it doesn't accept the pattern.
> 
> Is your pattern the standard 3x3, or is it one of the custom Lineage patterns like 4x4 or 5x5, etc? I noticed a similar issue with larger-than-standard patterns and TWRP decrypt not working.

No, just the standard 3x3 pattern.

**Comment by Herzfinsternis** [comment]
> Everything worked before with LO 14.1

This probably is an issue with Project Treble. In that case you will need a compatible recovery. I had the same issue when my device (mido) was updated from 14.1 to 15.1 and found a special TWRP version on xda-developers.com.

You could try the recovery linked from _[removed]_ which leads you to _[removed]_

But I do not own a Wileyfox Swift, so I cannot tell, if this will solve your problem or will give you even more headache.

**[UPDATE]**
Just found another on on https://forum.xda-developers.com/wileyfox-swift/orig-development/rom-lineageos-15-1-wileyfox-swift-t3792752
**[ROM][8.1.0][OFFICIAL] LineageOS 15.1 for Wileyfox Swift by jrior001**
**_Updated recovery required:_**
http://theflamingskull.com/downloads/crackling/TWRP-crackling-3.2.1.img
**This seems to be from the official maintainer!**

**Comment by TheAssassin** [comment]
**TL;DR: that recovery doesn't work either!**

I just tested that TWRP @Herzfinsternis linked to on a crackling device, but it can't decrypt my data partition either. I just reinstalled that phone with the latest LineageOS 15.1 build, and booting etc. works fine. I came here after trying all the official TWRP builds. I had to reinstall because the upgrade (dirty flash) via TWRP didn't work. Now, I can't install any more zips, as there's no working recovery.

This is quite annoying. 15.1 runs quite a bit better on that crackling than 14.1... but I want to install those zips again, which really improve LineageOS's UX.

I use a passphrase by the way, not a pattern (didn't know that was an option for encrypting the phone, but I consider patterns quite insecure anyway).

I get a feeling that either the decryption libraries in the TWRP build aren't up to date (according to the changelog, that stuff wasn't updated for a while). But I have never looked into TWRP more than as a user, so I can't tell.

**Comment by gyorgyabraham** [comment]
I experienced the same issue, using OmniROM 7.1.2. After install with TWRP, I encrypthed the whole phone with PIN code, which encrypts all internal data on the phone, rendering latest twrp release for athene unable to decrypt. It does not ask for password. I tried PIN, pattern, and password, and no password at all, it does not help.
I found an unofficially modified and built version of TWRP which immediately successfully asked for my PIN and decrypted partitions. This is required to do for example backups or OS upgrades with TWRP.
https://forum.xda-developers.com/moto-g4-plus/development/twrp-moto-g4-g4-plus-t3684813

EDIT: despite the fact that decryption was success (I could browse phone content via MTP USB mount on my Linux machine while running TWRP), something went wrong. I upgraded from Omni7 to Omni8.1 with encryption on. Upon first boot, the system asked for my PIN to decrypt, which was accepted but it told me my data is corrupted, and my only button to choose was to do a factory reset. I dont know the reason, maybe Omni7 and Omni8.1 uses a different type of encryption or so.
